### PR TITLE
clipper2: init at 1.2.2

### PR DESCRIPTION
**Pull request by cadkin**
###### Description of changes

[Repo Link](https://github.com/AngusJohnson/Clipper2)

Clipper2 is a major update to the Clipper C++ library by Angus Johnson. It provides polygon clipping functions and some other 2D geometry utilities. 

We have the original Clipper library in nixpkgs but not the new version, so let's get it added.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/clipper2/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake
+}:
```
Comment (AndersonTorres) [inline pkgs/development/libraries/clipper2/default.nix:6]:
```suggestion
{ lib
, stdenv
, fetchFromGitHub
, cmake
}:
```

File: pkgs/development/libraries/clipper2/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "Clipper2";
```
Comment (AndersonTorres) [inline pkgs/development/libraries/clipper2/default.nix:9]:
```suggestion
  pname = "clipper2";
```

File: pkgs/development/libraries/clipper2/default.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  version = "1.2.2";
+
+  src = fetchFromGitHub {
+    owner  = "AngusJohnson";
+    repo   = "${pname}";
+    rev    = "${pname}_${version}";
+    sha256 = "sha256-u/4GG1A2PAlk8VEWgJX8+EnZ5hpGhu1QbvHwct58sF4=";
+  };
```
Comment (AndersonTorres) [inline pkgs/development/libraries/clipper2/default.nix:17]:
```suggestion
  src = fetchFromGitHub {
    owner = "AngusJohnson";
    repo = "Clipper2";
    rev = "Clipper2_${version}";
    hash = "sha256-u/4GG1A2PAlk8VEWgJX8+EnZ5hpGhu1QbvHwct58sF4=";
  };
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by cadkin** [comment]
@AndersonTorres  Resolved formatting issues - take a look at your convenience.

**Review by AndersonTorres — APPROVED** [review]
Ping me 3 days and I'll merge this.

**Comment by cadkin** [comment]
@AndersonTorres Hey when you get a chance can you look into merging it again? Thanks!

File: pkgs/development/libraries/clipper2/0001-fix-pc-paths.patch [hunk 4]
```diff
@@ -0,0 +1,14 @@
+diff --git a/Clipper2.pc.cmakein b/Clipper2.pc.cmakein
+index 5632c93..0a5d6ba 100644
+--- a/Clipper2.pc.cmakein
```
Comment (AndersonTorres) [inline pkgs/development/libraries/clipper2/0001-fix-pc-paths.patch:3]: dunno if I commented this before, but
1. It should use [GNUInstallDirs](https://cmake.org/cmake/help/v3.27/module/GNUInstallDirs.html) module.
2. It should be upstreamed

Comment (cadkin) [reply pkgs/development/libraries/clipper2/0001-fix-pc-paths.patch:3]: I agree, I'll check if the upstream maintainers are good with it.

**Pull request merged**